wandb/data_types.py

```diff
@@ -11,11 +36,40 @@
 from wandb import util
 
 
-def val_to_json(key, val, mode="summary", step=None):
-    """Converts a wandb datatype to its JSON representation"""
+# Staging directory so we can encode raw data into files, then hash them before
+# we put them into the Run directory to be uploaded.
+MEDIA_TMP = tempfile.TemporaryDirectory('wandb-media')
+
+
+DATA_FRAMES_SUBDIR = os.path.join('media', 'data_frames')
+
+
+def history_dict_to_json(run, payload, step=None):
+    """Converts a History row dict's elements so they're friendly for JSON serialization.
+    """
+    if step is None:
+        # We should be at the top level of the History row; assume this key is set.
+        step = payload['_step']
+
+    for key, val in six.iteritems(payload):
+        if isinstance(val, dict):
+            payload[key] = history_dict_to_json(run, val, step=step)
+        else:
+            payload[key] = val_to_json(run, key, val, step=step)
+
+    return payload
+
+
+def val_to_json(run, key, val, step='summary'):
+    """Converts a wandb datatype to its JSON representation.
+    """
     converted = val
     typename = util.get_full_typename(val)
-    if util.is_matplotlib_typename(typename):
+
+    if util.is_pandas_data_frame(val):
+        assert step == 'summary', "We don't yet support DataFrames in History."
```

Comment:
Maybe say we don't support dataframes in calls to `wandb.log`.  Users won't know what History is.

Reply:
Good point.